presto-manager-tests/src/main/java/com/teradata/prestomanager/tests/TestBasicFunction.java

```diff
@@ -0,0 +1,314 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.tests;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.glassfish.jersey.client.JerseyClientBuilder;
+import org.testng.annotations.Test;
+
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly;
+import static java.util.concurrent.Executors.newSingleThreadExecutor;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test
+public class TestBasicFunction
+{
+    private final URI controllerUri;
+    private final URI coordinatorUri;
+    private final Client client;
+    private final int numPrestoNodes;
+
+    //TODO: Change to use TESTNG Annotations i.e @beforeClass
+    public TestBasicFunction()
+            throws URISyntaxException
+    {
+        controllerUri = new URI("http://localhost:8088");
+        coordinatorUri = new URI("http://localhost:8080");
+        client = JerseyClientBuilder.createClient();
+        numPrestoNodes = 2;
+    }
+
+    //TODO: Replace this hack of checking if presto manager is up
+    // and discovery service has kicked in
+    private boolean check_prestoManager()
+    {
+        List<JsonObject> statusList;
+        try {
+            statusList = parseStatusResponse(queryPrestoStatus());
+        }
+        catch (ProcessingException e) {
+            return false;
+        }
+        return statusList.size() == numPrestoNodes;
+    }
+
+    @Test
+    public void runAll()
+            throws IOException, InterruptedException, TimeoutException, ExecutionException
+    {
+        retry(this::check_prestoManager, true, 1, 300);
+
+        System.out.println("TEST STARTS HERE");
```

Comment:
you shouldn't print stuff out during the test.  If it fails somewhere, you'll see that.